scene/gui/color_picker.cpp

```diff
@@ -1053,6 +1058,9 @@ void ColorPicker::_screen_input(const Ref<InputEvent> &p_event) {
 			Color c = img->get_pixel(ofs.x, ofs.y);
 
 			set_pick_color(c);
+			if (!deferred_mode_enabled) {
+				emit_signal(SNAME("color_changed"), color);
+			}
```

Comment:
This makes the `ColorPickerButton`'s color keep changing while sampling. This will behave differently than before, but I'm not sure which is the desired behavior.

Reply:
It does indeed. As far as I could see, using the screen picker was the only case where the pick color was changed without also updating the ColorPickerButton, which I felt was a bit inconsistent. I started working on some refactoring of ColorPicker to avoid code duplication and inconsistencies, but then I learned about Vitika's GSoC project so I put that on ice.

It's not exactly the same, but when sampling the UV circle it continuously updates the ColorPickerButton (unless deferred mode is active).

I suggested this modification to KoBeWi on discord and they didn't seem opposed to the change in behaviour at least. Don't know if there's any formal procedure changing functionality that I missed.

Reply:
It may be considered the timing of the `color_changed` signal in `ColorPickerButton`. It feels that the previous implementation hopes to emit the `color_changed` signal when the sampling ends. In contrast to other cases, sampling is an action with a clear end flag, so it might make sense to treat the color as a temporary color during sampling and not emit it.

Reply:
Yep, I see the logic in treating screen picking as a singular event that happens when the mouse is clicked, and I see the need for displaying the currently hovered color to the user. It's just that IMO the purpose of the "sample" (i.e. the swatch that displays the old and current colors) is a bit ambivalent. Most of the time the color that is shown on the sample is actually chosen, and events are emitted, except for when using the screen picker where it changes to display a temporary color that isn't chosen.

But maybe it would be better to go the opposite way. If the user has grabbed a slider handle, or is holding LMB on the UV circle, have they really selected a color before releasing the mouse button? If you're dragging the slider back and forth to find the color you're looking for, isn't that kind of temporary too? The sample should ofc be continuously be updated to show the current color, but maybe color_changed should only be emitted when the action ends? Which I think is basically deferred_mode..

I would be content with either behaviour, as long as it is consistent :)